core/renderers/minimalist/info.ts

```diff
@@ -28,6 +28,9 @@ import type {Renderer} from './renderer.js';
  * @alias Blockly.minimalist.RenderInfo
  */
 export class RenderInfo extends BaseRenderInfo {
+  // Exclamation is fine b/c this is assigned by the super constructor.
+  protected override renderer_!: Renderer;
```

Comment:
Then why does it need to be overridden here at all, instead of being `!`ed in `BaseRenderInfo`?

Reply:
Here `Renderer` is the minimalist renderer, not the common renderer. So we're actually overriding it to change the type. We just also need to do the assertion b/c typescript can't figure out that its being set properly.